### ISSUE TITLE
Broken with WP 4.8.2 & 4.7.6

### ISSUE THREAD

**Issue by tlhackque** [KingYes/wordpress-aryo-activity-log#102]
After upgrading to WP 4.8.2, activity is logged in the database, but the display reports "no items found:".

Multiple reports - see https://wordpress.org/support/topic/no-items-found-6/

**Comment by verne-wv** [comment]
I know my v4.7.6 needs upgrading, but wanted to report I too am seeing this issue.

**Comment by goatherd** [comment]
There are a couple of failed SQL log entries caused by the security fix on WP 4.8.2.

```
017/09/20 19:29:46 [error] 22238#22238: *12935232 FastCGI sent in stderr: "PHP message: WordPress database error Table 'wp.%1$s' doesn't exist for query SELECT COUNT(`histid`) FROM `%1$s`
				 WHERE 1=1
					AND (`object_type` = 'Core' OR `object_type` = 'Export' OR `object_type` = 'Post' OR `object_type` = 'Taxonomy' OR `object_type` = 'User' OR `object_type` = 'Options' OR `object_type` = 'Attachment' OR `object_type` = 'Plugin' OR `object_type` = 'Widget' OR `object_type` = 'Theme' OR `object_type` = 'Menu' OR `object_type` = 'Comments') AND (`user_caps` = 'administrator' OR `user_caps` = 'editor' OR `user_caps` = 'author' OR `user_caps` = 'guest') made by do_action('toplevel_page_activity_log_page'), WP_Hook->do_action, WP_Hook->apply_filters, call_user_func_array, AAL_Admin_Ui->activity_log_page_func, AAL_Activity_Log_List_Table->prepare_items
PHP message: WordPress database error You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '%4$d, %5$d' at line 5 for query SELECT * FROM `%1$s`
				 WHERE 1=1
					AND (`object_type` = 'Core' OR `object_type` = 'Export' OR `object_type` = 'Post' OR `object_type` = 'Taxonomy' OR `object_type` = 'User' OR `object_type` = 'Options' OR `object_type` = 'Attachment' OR `object_type` = 'Plugin' OR `object_type` = 'Widget' OR `object_type` = 'Theme' OR `object_type` = 'Menu' OR `object_type` = 'Comments') AND (`user_caps` = 'administrator' OR `user_caps` = 'editor' OR `user_caps` = 'author' OR `user_caps` = 'guest')
					ORDER BY `%2$s` %3$s
					LIMIT %4$d, %5$d; made by do_action('toplevel_page_activity_log_page'), WP_Hook->do_action, WP_Hook->apply_filters, call_user_func_array, AAL_Admin_Ui->activity_log_page_func, AAL_Activity_Log_List_Table->prepare_items
PHP message: WordPress database error You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '%1$s FROM `%2$s`
				WHERE 1 = 1
				AND (`object_type
```

```
2017/09/20 19:29:46 [error] 22238#22238: *12935232 FastCGI sent in stderr: "ply_filters, call_user_func_array, AAL_Admin_Ui->activity_log_page_func, WP_List_Table->display, AAL_Activity_...
PHP message: WordPress database error You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '%1$s FROM `%2$s`
				WHERE 1 = 1
				AND (`object_type` = 'Core' OR `object_type' at line 1 for query SELECT DISTINCT %1$s FROM `%2$s`
				WHERE 1 = 1
				AND (`object_type` = 'Core' OR `object_type` = 'Export' OR `object_type` = 'Post' OR `object_type` = 'Taxonomy' OR `object_type` = 'User' OR `object_type` = 'Options' OR `object_type` = 'Attachment' OR `object_type` = 'Plugin' OR `object_type` = 'Widget' OR `object_type` = 'Theme' OR `object_type` = 'Menu' OR `object_type` = 'Comments') AND (`user_caps` = 'administrator' OR `user_caps` = 'editor' OR `user_caps` = 'author' OR `user_caps` = 'guest')
				GROUP BY `%1$s`
				ORDER BY `%1$s`
			; made by do_action('toplevel_page_activity_log_page'), WP_Hook->do_action, WP_Hook->apply_filters, call_user_func_array, AAL_Admin_Ui->activity_log_page_func, WP_List_Table->display, AAL_Activity_" while reading response header from upstream, request: "GET /wp-admin/admin.php?page=activity_log_page HTTP/1.1
```

And similar reports around the `%1$s`-syntax on event logging calls (login-failed being the most frequent on our side with several dozen of entries every minute).

**Comment by EiraStudios** [comment]
This issue is fixed with the following pull request: #104

**Comment by tlhackque** [comment]
@EiraStudios: Thank you.  

I can confirm that this fixes the issues with 4.8.2.  Note to other who merge manually: only the first two changed files exist in a normal installation; the changes in `tests/test-base.php` only apply to developers.

If you're in a hosted WP environment, the easiest thing to do is to use the plugin editor to completely replace the two files.  Visit https://github.com/KingYes/wordpress-aryo-activity-log/pull/106/files.  Click "VIEW" for each file, then "RAW"; copy and paste into the plugin.  You may need to add a newline to the pasted content.

Note that if you're running wordfence (or another monitor), you'll get a notice that the plugin differs from the distribution.

Of course, backup your site (or at least these two files) first!  And update to the released version as soon as possible.

Now if someone could take on fixing #101 ....

**Comment by KingYes** [comment]
Merge by #106 

Thanks guys
